### PR TITLE
Fixes Unique check for event creation bug

### LINKED ISSUE #690: [BUG] Unique check for event creation doesnt take chapter into consideration
### Describe the bug

When an event is created under a FOSS City Chapter (like /c/mumbai/nov24), only the **nov24** part is considered in the unique check logic, meaning that no 2 chapters can have the same slug.

### Expected behavior

Allow multiple city chapters to have same slug like /nov24

### To Reproduce

Create an event with the "Event Permalink" set to nov24

### Screenshots

image
In this case, the event permalink is same for mumbai (which I was trying to create), and BLR (which was already created)

### Environment

```markdown
- Device: Desktop
- OS: Debian
- Browser: Zen Browser (Firefox)
```

### PR DESCRIPTION
**Pull request by vasujain275**
## What type of PR is this? (check all applicable)

- [ ] 🍕Feature
- [ ] ⚙️Chore
- [x] 🐛Bug Fix
- [ ] 🎨Style
- [ ] 🧑‍💻Refactor
- [ ] 📕Documentation
- [ ] 🏎️Optimization

## Description

Unchecked the Unique flag on `Event Permalink` from `FOSS Chapter Event` Doctype. Now, two different chapters can have same slug in event permalink. Also modified test case to validate the same.

## Related Issues & Docs

closes #690 

## Checklist

- [x] I have read the [contribution guidelines]("./docs/CONTRIBUTING.md").
- [x] I have tested these changes locally.
- [ ] I have updated the documentation (If applicable).
- [x] The code follows the project's coding standards.
- [x] I have added/updated tests, if applicable.

## Screenshots/GIFs/Screen Recordings (if applicable)

N/A

## Additional context

N/A

### CODE REVIEW

**Comment by vasujain275** [comment]
Sorry for the linting/formatting errors, reinstalled the local bench and forgot to install pre-commit hook

**Comment by rahulporuri** [comment]
i sincerely apologize but i have a fundamental question about this approach - https://github.com/fossunited/fossunited/issues/690#issuecomment-2484626162

**Comment by harshtandiya** [comment]
Looks like your precommit is not set up correctly @vasujain275

**Comment by vasujain275** [comment]
@harshtandiya my bad, now pre commit is working on my local setup correctly

File: fossunited/chapters/doctype/foss_chapter_event/test_foss_chapter_event.py [hunk 1]
```diff
@@ -107,29 +107,64 @@ def test_members_are_added_to_event(self):
...
+                    "chapter": chapter_name,
+                    "event_name": fake.text(max_nb_chars=40),
+                    "event_type": "FOSS Meetup",
+                    "event_permalink": event_permalink,
+                    "status": "Live",
+                    "event_start_date": datetime.now() + timedelta(days=4),
+                    "event_end_date": datetime.now() + timedelta(days=5),
+                    "event_description": fake.text(max_nb_chars=200),
+                }
+            )
```
Comment (harshtandiya) [inline fossunited/chapters/doctype/foss_chapter_event/test_foss_chapter_event.py]: Alignment isssues? Dont think this is supposed to be here. 

I do like the idea of having central functions for creating chapter and event. However, I do not think we should do it here. We can create a `utils.py` file in the test directory inside fossunited. In that file we can write these functions and then import them to other tests to use.

File: fossunited/chapters/doctype/foss_chapter_event/test_foss_chapter_event.py [hunk 2]
```diff
@@ -107,29 +107,64 @@ def test_members_are_added_to_event(self):
...
+                    "slug": fake.slug(),
+                    "city": "Kochi",
+                    "country": "India",
+                    "email": fake.email(),
+                    "facebook": fake.url(),
+                    "instagram": fake.url(),
+                    "linkedin": fake.url(),
+                    "mastodon": fake.url(),
+                    "matrix": fake.url(),
+                    "state": "Kerala",
```
Comment (harshtandiya) [inline fossunited/chapters/doctype/foss_chapter_event/test_foss_chapter_event.py]: same as above

File: fossunited/chapters/doctype/foss_chapter_event/foss_chapter_event.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2024, Frappe x FOSSUnited and contributors
+// For license information, please see license.txt
+
+// frappe.ui.form.on("FOSS Chapter Event", {
+// 	refresh(frm) {
+
+// 	},
+// });
```
Comment (harshtandiya) [inline fossunited/chapters/doctype/foss_chapter_event/foss_chapter_event.js]: Let's unstage this file

File: fossunited/chapters/doctype/foss_chapter_event/foss_chapter_event.json [hunk 4]
```diff
@@ -1,549 +1,548 @@
...
-      "color": "Gray",
-      "title": "Concluded"
-    },
-    {
-      "color": "Red",
-      "title": "Cancelled"
-    }
-  ],
-  "title_field": "event_name"
+ "actions": [],
```
Comment (harshtandiya) [inline fossunited/chapters/doctype/foss_chapter_event/foss_chapter_event.json]: `yarn format` will fix this file's formatting

**Review by harshtandiya — CHANGES_REQUESTED** [review]

**Review by harshtandiya — APPROVED** [review]
great job!

**Pull request merged**